platform/default/mbgl/storage/offline_database.cpp

```diff
@@ -458,6 +466,7 @@ void OfflineDatabase::deleteRegion(OfflineRegion&& region) {
     stmt->run();
 
     evict(0);
+    db->exec("PRAGMA incremental_vacuum");
```

Comment:
I think calling vacuum should be a separated API and not called automatically. I can think of 2 use cases for that:
 - In your app you want to remove several regions. Each deletion would be super slow because the database would be recreated on each VACUUM. Instead, the interface would delete first and at the end call `OffilineDatabase::vacuum()`
 - This API could be used for vacuum the dangling pages from the cache.